### PR TITLE
Overlay 124

### PR DESCRIPTION
**Pull request by PikalaxALT**
i have a terrible habit of making PRs depend on what came before

### CODE REVIEW

File: include/overlay_123.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#ifndef GUARD_POKEHEARTGOLD_OVERLAY_123_H
+#define GUARD_POKEHEARTGOLD_OVERLAY_123_H
+
+#include "global.h"
+
+BOOL ov123_0225F4A8(void (*)(void));
```
Comment (adrienntindall) [inline include/overlay_123.h:6]: Ideally we typedef a function pointer here

File: src/overlay_124.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    fieldSystem->bagCursor = BagCursor_New(HEAP_ID_FIELD);
+    FS_UnloadOverlay(MI_PROCESSOR_ARM9, FS_OVERLAY_ID(OVY_123));
+
+    // all combinations of the three prime multipliers above are coprime with 2441 and 4073
+    if ((key + key2) % 2441) {
+        ov124_02260D1C(fieldSystem);
+    }
+    fieldSystem->unkA8 = sub_02092BB8(HEAP_ID_FIELD);
+    fieldSystem->unk108 = FieldSystem_UnkSub108_Alloc(HEAP_ID_FIELD);
+    fieldSystem->unk114 = GearPhoneRingManager_New(HEAP_ID_FIELD, fieldSystem);
```
Comment (adrienntindall) [inline src/overlay_124.c:40]: can name unk114

File: src/overlay_124.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    int numBadges = 0;
+    for (i = 0; i < 16; ++i) {
+        if (PlayerProfile_TestBadgeFlag(Save_PlayerData_GetProfileAddr(fieldSystem->saveData), i) == TRUE) {
+            ++numBadges;
+        }
+    }
+    AllocFromHeapAtEnd(HEAP_ID_3, numBadges == 0 ? 20000 : (20000 * numBadges));
+}
+
+// you can have a little heap clobber, as a treat
```
Comment (adrienntindall) [inline src/overlay_124.c:59]: these comments are funny but prolly not needed

**Review by adrienntindall — CHANGES_REQUESTED** [review]
pretty much just one naming thing otherwise looks good

**Review by tgsm — DISMISSED** [review]
Looks fine

File: include/overlay_123.h (hunk 1 above)
Comment (PikalaxALT) [reply include/overlay_123.h:6]: i mean, it's overlay123 which is the biggest questionmark in all of johto. leaving as-is

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**